system_architecture_diagram.md

```diff
@@ -0,0 +1,175 @@
+
+### 🟣 Evaluation Pipeline (Medium Gray)
+- **7 Active Tasks**: Translation (bidirectional), Classification, MMLU, ARC, TruthfulQA, MGSM
+- **Unified English Zero-Shot Prompting**: All tasks use English instructions with target language content
+- **Origin Tagging**: Distinguishes between human-translated ('human') and machine-translated ('machine') data
+- **Combinatorial Approach**: Systematic evaluation across Model × Language × Task combinations
+- **Sample-based**: 10 evaluations per combination for statistical reliability
+- **Batch Processing**: 50 tasks per batch with rate limiting and error resilience
+- **Dual Deployment**: `main.py` for local/GitHub, `main_gcs.py` for Google Cloud with GCS storage
+
+### 🟠 API Integration (Light Gray)
+- **OpenRouter**: Primary model inference API for all language model tasks
+- **Rate Limiting**: Intelligent batching and delays to prevent API overload
+- **Error Handling**: Graceful handling of timeouts, rate limits, and model unavailability
+- **HuggingFace**: Model metadata and open-source model information
+- **Google Translate**: Specialized translation API for on-the-fly dataset translation
+
+### 🟢 Data Storage (Cyan)
+- **results.json**: Aggregated evaluation scores with origin-specific metrics
+- **models.json**: Dynamic model list with metadata and validation status
+- **languages.json**: Language information with population data
+
+### 🟡 Frontend Visualization (Light Red)
+- **WorldMap**: Interactive country-level language proficiency visualization
+- **ModelTable**: Ranked model performance leaderboard with origin-specific columns
+- **LanguageTable**: Language coverage and speaker statistics
+- **DatasetTable**: Task-specific performance breakdowns with human/machine distinction
+
+### 🔵 Translation & Origin Tracking (Light Green)
+- **On-the-fly Translation**: Google Translate API for languages without native benchmarks
+- **Origin Tagging**: Automatic classification of data sources (human vs. machine translated)
+- **Separate Metrics**: Frontend displays distinct scores for human and machine-translated data
+
+## Data Flow Summary
+
+1. **Model Discovery**: Combine curated + trending models → validate API availability → enrich with metadata
+2. **Evaluation Setup**: Generate all valid Model × Language × Task combinations with origin tracking
+3. **Task Execution**: Run evaluations using unified English prompting and appropriate datasets
+4. **Result Processing**: Aggregate scores by model+language+task+origin and save to JSON files
+5. **Backend Serving**: FastAPI serves processed data with origin-specific metrics via REST API
+6. **Frontend Display**: React app visualizes data through interactive components with transparency indicators
+
+This architecture enables scalable, automated evaluation of AI language models across diverse languages and tasks while providing real-time insights through an intuitive web interface with methodological transparency. 
```

Comment:
maybe move into notes/ folder (which we could rename to docs/)
maybe mention at the top that it's not 100% up to date